eras/shelley/impl/src/Cardano/Ledger/Shelley/Rules/Upec.hs

```diff
@@ -54,6 +56,17 @@ import qualified Data.Map.Strict as Map
 import GHC.Generics (Generic)
 import NoThunks.Class (NoThunks (..))
 
+data UpecState era = UpecState
+  { currentPp :: !(PParams era)
+  -- ^ Current protocol parameters.
+  , ppupState :: !(ShelleyPPUPState era)
+  -- ^ State of the protocol update transition system.
+  }
```

Comment:
nice! I wonder how this found itself living in the ledger state module... :thinking:

Reply:
I was wondering the same thing :smile: